tests/test_cache_memoize.py

```diff
@@ -62,7 +69,19 @@ def returnnothing(a, b, k="bla"):
 
     returnnothing(1, 2)
     returnnothing(1, 2)
-    assert len(calls_made) == 8
+    assert len(calls_made) == 1
+
+
+def test_colons():
+    calls_made = []
+    @cache_memoize(10)
+    def fun(a, b, k="bla"):
+        calls_made.append((a, b, k))
+        return (a, b, k)
+
+    fun('a:b', 'c')
+    fun('a', 'b:c')
+    assert len(calls_made) == 2
```

Comment:
Can you extend this unit test to do one with non-ascii parameters.
E.g.
```python
fun('a', 'ä:á')
```
or whatever it is. 

Also, what would happen if you pass byte string instead. E.g. `b'ë'`

Reply:
Sure thing.